ui/src/containers/Layout.js

```diff
@@ -21,139 +22,167 @@ import { updateLanguageAction } from '../ducks/config';
+    }
+  ];
 
-    const user = {
-      name: this.props.user && this.props.user.username,
-      actions: [
-        { label: this.props.intl.messages.log_out, onClick: this.props.logout }
-      ]
-    };
-
-    const sidebar = {
-      expanded: this.props.sidebar.expanded,
-      actions: [
-        {
-          label: this.props.intl.messages.monitoring,
-          icon: <i className="fas fa-desktop" />,
-          onClick: () => {
-            this.props.history.push('/');
-          },
-          active: matchPath(this.props.history.location.pathname, {
-            path: '/',
-            exact: true,
-            strict: true
-          })
-        },
-        {
-          label: this.props.intl.messages.nodes,
-          icon: <i className="fas fa-server" />,
-          onClick: () => {
-            this.props.history.push('/nodes');
-          },
-          active: matchPath(this.props.history.location.pathname, {
-            path: '/nodes',
-            exact: false,
-            strict: true
-          })
-        }
-      ]
-    };
+  const user = {
+    name: props.user && props.user.username,
+    actions: [{ label: props.intl.messages.log_out, onClick: props.logout }]
+  };
 
-    // In this particular case, the label should not be translated
-    const languages = [
+  const sidebar = {
+    expanded: props.sidebar.expanded,
+    actions: [
       {
-        label: 'Français',
-        name: FR_LANG,
+        label: props.intl.messages.monitoring,
+        icon: <i className="fas fa-desktop" />,
         onClick: () => {
-          this.props.updateLanguage(FR_LANG);
+          props.history.push('/');
         },
-        selected: this.props.language === FR_LANG,
-        'data-cy': FR_LANG
+        active: matchPath(props.history.location.pathname, {
+          path: '/',
+          exact: true,
+          strict: true
+        })
       },
       {
-        label: 'English',
-        name: EN_LANG,
+        label: props.intl.messages.nodes,
+        icon: <i className="fas fa-server" />,
         onClick: () => {
-          this.props.updateLanguage(EN_LANG);
+          props.history.push('/nodes');
         },
-        selected: this.props.language === EN_LANG,
-        'data-cy': EN_LANG
+        active: matchPath(props.history.location.pathname, {
+          path: '/nodes',
+          exact: false,
+          strict: true
+        })
       }
-    ];
+    ]
+  };
 
-    const navbar = {
-      onToggleClick: this.props.toggleSidebar,
-      toggleVisible: true,
-      productName: this.props.intl.messages.product_name,
-      applications,
-      help,
-      user: this.props.user && user,
-      languages,
-      logo: (
-        <img
-          alt="logo"
-          src={process.env.PUBLIC_URL + '/brand/assets/branding.svg'}
-        />
-      )
-    };
+  let applications = null;
+  if (props.solutions && props.solutions.length) {
+    applications = [];
+    props.solutions.reduce((applications, solution) => {
```

Comment:
This transformation is a bit confusing since you are using reduce and app but what actually really create you list is the `push`.
Maybe you can use one of those implementation : 
```javascript
    const navbarApps = props?.solutions?.reduce((prev, solution) => {
      let deployedVersionsSolution = [];
      solution?.versions?.forEach(version => {
        if (version?.deployed && version?.ui_url) {
          deployedVersionsSolution.push({
            label: solution.name,
            // TO BE IMPROVED in core-ui to allow display Link or <a></a>
            onClick: () => window.open(version.ui_url, '_self')
          });
        }
      });
      return [...prev, ...deployedVersionsSolution];
    }, []);

    const navbarApp2 = props?.solutions?.reduce((prev, solution) => {
      let solutionDeployedVersions = solution?.versions?.filter(
        version => version?.deployed && version?.ui_url
      );
      let app = solutionDeployedVersions.map(version => ({
        label: solution.name,
        // TO BE IMPROVED in core-ui to allow display Link or <a></a>
        onClick: () => window.open(version.ui_url, '_self')
      }));
      return [...prev, ...app];
    }, []);
```

An implementation with only forEach is possible too.
Feel free to improve the way that is the more natural for you !